### PR TITLE
RGB24: decentralized global naming

### PR DESCRIPTION
**Pull request by dr-orlovsky**
This is the last bit in what I was considering a valid use cases for RGB. The rest - BiFi - would be actually a layer on top of RGB, composable of RGB20-24 schemata with nothing more needed.

RGB24 - a better & valid analog of ENS, solving problems of DNS, the problems ENS was not able to solve - is the checkpoint for myself on RGB and Contractum readiness.

For two years I was not able to write a valid code in terms of RGB which would be able to express business logic necessary to run that sort of solution.

Finally, I was able to do that. I understand that it would be probably quite hard to understand what the code means - but this is so far is the state of the art what can be achievable with Contractum and RGB at all. I tried to make it as readable as possible.

The code demonstrates:
- inter-contract interactions
- in a nested (tree-like) way
- with reflection back to the parent contract
- and cross-contract validation
- plus introspection on own contract properties

Basically, this is a recursive RGB contract interacting with other contracts with two-way recursion.

PS. Just compare the amount of code here with the amount of code in the ENS!!!

### CODE REVIEW

File: lnpbp-0024.md [hunk 1]
```diff
@@ -40,6 +43,23 @@ License: CC0-1.0
 Interface specification is the following Contractum code:
 
 ```haskell
+interface RGB24
+    global Root :: Self.ContractId?
+    global Name :: Ident
+    global Registry :: {Ident -> Resolve}
```
Comment (dr-orlovsky) [inline lnpbp-0024.md:49]:
```suggestion
    global Registry { Ident } :: Resolve
```

File: lnpbp-0024.md [hunk 2]
```diff
@@ -59,4 +103,4 @@ Interface specification is the following Contractum code:
 
 ## Copyright
 
-This document is licensed under the Creative Commons CC0 1.0 Universal license.
+This document is licensed under the GPL-3.0 license.
```
Comment (jirijakes) [inline lnpbp-0024.md:106]: This left me puzzled – can you please describe motivations behind the change to GPL-3.0? Perhaps in the PR itself. Or is there some related discussion about this? Wouldn't it imply (Section 5) that much more code that builds on top of RGB24 would have to be distributed under GPL3? Thank you.

Comment (dr-orlovsky) [reply lnpbp-0024.md:106]: The rationale is simple: a global naming system can't be a non-open by a definition, meaning that it's data must be available upon a request.

This standard describes an API - and not an implementation - of a such system. Thus software running it can operate on any license, while API has to be opened.
